packages/flutter_test/lib/src/_test_selector_web.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+import 'dart:ui' as ui;
+import 'dart:ui_web' as ui_web;
+
+import 'package:stream_channel/stream_channel.dart';
+import 'package:test_api/backend.dart';
+
+import '_goldens_web.dart';
+import 'goldens.dart';
+import 'web.dart' as web;
+
+// This file contains APIs that are used by the generated test harness for
+// running flutter unit tests.
+
+/// A `main` entry point for a test.
+typedef EntryPoint = FutureOr<void> Function();
+
+/// An entry point runner provided by a test config file
+typedef EntryPointRunner = Future<void> Function(EntryPoint);
+
+/// Metadata about a web test to run
+typedef WebTest = ({
+  EntryPoint entryPoint,
+  EntryPointRunner? entryPointRunner,
+  Uri goldensUri,
+});
+
+/// Gets the test selector set by the test bootstrapping logic
+String get testSelector {
+  final JSString? jsTestSelector = web.window.testSelector;
+  if (jsTestSelector == null) {
+    throw Exception('Test selector not set');
+  }
+  return  jsTestSelector.toDart;
+}
+
+/// Runs a specific web test
+Future<void> runWebTest(WebTest test) async {
+  ui_web.debugEmulateFlutterTesterEnvironment = true;
+  final Completer<void> completer = Completer<void>();
+  await ui_web.bootstrapEngine(runApp: () => completer.complete());
+  await completer.future;
+  webGoldenComparator = DefaultWebGoldenComparator(test.goldensUri);
+
+  /// This hard-codes the device pixel ratio to 3.0 and a 2400 x 1800 window
+  /// size for the purposes of testing.
+  ui_web.debugOverrideDevicePixelRatio(3.0);
+  ui.window.debugPhysicalSizeOverride = const ui.Size(2400, 1800);
```

Comment:
fly-by comment: The `window` singleton is deprecated and we've cleaned up the code base to no longer depend on this singleton. Can you please not introduce new references to it?

Reply:
This is not a new reference, it is copied over from the previous generated test bootstrap file. I'm happy to switch over to a different API though, as long as it is guaranteed to have the exact same behavior. What should I invoke instead?

Reply:
I did not find an equivalent API that is accessible from this package. We can look into figuring out a solution for this later. Especially since this isn't really a new reference to this API, I don't want to block this change on it.

Reply:
I've filed https://github.com/flutter/flutter/issues/145558 to keep track of this.